### PR TITLE
Schedule calendar notifications on device restart

### LINKED ISSUE #4245: Re-schedule calendar notifications after device Restart
This issue depends on #4244.

Android is nice to us and removes all pending intents when a device is restarted, we've got to reschedule all pending intents using the `AlarmManager` once again when the device is restarted.

Hints:
Take a look at
```
<uses-permission android:name="android.permission.WAKE_LOCK" />
<uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
```

We are most likely going to create a broadcast receiver to listen to the device boot and re-schedule pending notifications.

**The notifications will have to be stored somewhere**, or we can use `ScheduleEventNotificationsInteractor` in the broascast receiver, if possible (this would actually be more precise, in case the events changed).

### PR DESCRIPTION
**Pull request by juanlaube**
## What's this?
This schedules the calendar notifications when the device is restarted.

### Relationships

Closes #4245 

## Why do we want this?
Because we want users to receive notifications, even if they restart their devices.

## How is it done?
It's a broadcast receiver that listens for `BOOT_COMPLETED` and starts a `JobIntentService` that triggers the interactor to schedule notifications.

### Why not another way?
There's no other way.

### Side effects
None.

## Review hints
I left some logs for debugging and testing purposes, you can restart a device (or the emulator) and see the logs with `adb logcat | grep asdf`

## :squid: Permissions
🚫 Don't, I will remove all the logs after someone tested this.

### CODE REVIEW

File: Toggl.Giskard/BroadcastReceivers/BootCompletedBroadcastReceiver.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+using Android.Support.V4.App;
+using Android.Util;
+using Toggl.Giskard.Services;
+
+namespace Toggl.Giskard.BroadcastReceivers
+{
+    [BroadcastReceiver(
+        Exported = true,
+        Permission = "android.permission.BIND_JOB_SERVICE",
+        Name = "com.toggl.giskardBootCompletedBroadcastReceiver")]
```
Comment (heytherewill) [inline Toggl.Giskard/BroadcastReceivers/BootCompletedBroadcastReceiver.cs]:
```suggestion
        Name = "toggl.giskard.BootCompletedBroadcastReceiver")]
```

**Comment by daividssilverio** [comment]
Also, please make `ScheduleEventNotificationsService` call `StopSelf` in case of an unhandled exceptions. We don't the app to keep crashing in case something fails horribly and the job intent service is retried.

It's alright not to call it if handle all possible exceptions though.

File: Toggl.Giskard/Services/ScheduleEventNotificationsService.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace Toggl.Giskard.Services
+{
+    [Service(
+        Exported = true,
+        Permission = "android.permission.BIND_JOB_SERVICE", 
+        Name = "com.toggl.giskard.ScheduleEventNotificationsService")]
+    public class ScheduleEventNotificationsService : JobIntentService
+    {
+        private IDisposable disposable;
```
Comment (daividssilverio) [inline Toggl.Giskard/Services/ScheduleEventNotificationsService.cs]: dispose this on `OnDestroy`.

Comment (daividssilverio) [reply Toggl.Giskard/Services/ScheduleEventNotificationsService.cs]: and set it to null there.

File: Toggl.Giskard/Services/ScheduleEventNotificationsService.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+        protected override void OnHandleWork(Intent intent)
+        {
+            var setup = MvxAndroidSetupSingleton.EnsureSingletonAvailable(ApplicationContext);
+            setup.EnsureInitialized();
+
+            if (!Mvx.TryResolve<IInteractorFactory>(out var interactorFactory))
+                return;
+
+            if (disposable != null)
```
Comment (daividssilverio) [inline Toggl.Giskard/Services/ScheduleEventNotificationsService.cs]: you could just use disposable?.Dispose() here as well.

File: Toggl.Giskard/Services/ScheduleEventNotificationsService.cs (hunk 2 above)
Comment (juanlaube) [reply Toggl.Giskard/Services/ScheduleEventNotificationsService.cs]: Done

**Pull request merged**